dbt/adapters/dremio/api/cursor.py

```diff
@@ -147,11 +147,31 @@ def _populate_rowcount(self):
 
         self._rowcount = rows
 
-    def _populate_job_results(self):
+    def _populate_job_results(self, row_limit=100):
         if self._job_results == None:
-            self._job_results = job_results(
-                self._parameters, self._job_id, offset=0, limit=500, ssl_verify=True
+            combined_job_results = job_results(
+                self._parameters,
+                self._job_id,
+                offset=0,
+                limit=row_limit,
+                ssl_verify=True,
             )
+            total_row_count = combined_job_results["rowCount"]
```

Comment:
Is there ever a case where "rowCount" isn't available for a job?

Reply:
_populate_job_results() will only execute after _populate_rowcount(). In _populate_rowcount() if rowCount is not in the job_status_response, it means that no rows will be returned, and rowCount will be set to -1. 

I have manually verified that this does not affect pagination.